common/src/main/java/io/pravega/common/concurrent/Futures.java

```diff
@@ -567,6 +567,47 @@
         result.whenComplete((r, ex) -> sf.cancel(true));
         return result;
     }
+    
+    /**
+     * Creates a new CompletableFuture that either holds the result of future from the futureSupplier
+     * or will timeout after the given amount of time.
+     *
+     * @param futureSupplier  Supplier of the future. 
+     * @param timeout         The timeout for the future.
+     * @param executorService An ExecutorService that will be used to invoke the timeout on.
+     * @param <T>             The Type argument for the CompletableFuture to create.
+     * @return A CompletableFuture which is either completed within given timebound or failed with timeout exception.
+     */
+    public static <T> CompletableFuture<T> futureWithTimeout(Supplier<CompletableFuture<T>> futureSupplier,
```

Comment:
please rewrite the `futureWithTimeout` above to make use of this method.
Ex:
`return futureWithTimeout(CompletableFuture::new, timeout, executorService)`